akka-cluster-sharding/src/multi-jvm/scala/akka/cluster/sharding/ClusterShardingSpec.scala

```diff
@@ -87,18 +88,21 @@ object ClusterShardingSpec {
     case ShardRegion.StartEntity(id) ⇒ (id.toLong % numberOfShards).toString
   }
 
-  def qualifiedCounterProps(typeName: String): Props =
-    Props(new QualifiedCounter(typeName))
+  def qualifiedCounterPropsFactory(typeName: String): String ⇒ Props =
+    id ⇒ Props(new QualifiedCounter(typeName, id))
 
-  class QualifiedCounter(typeName: String) extends Counter {
-    override def persistenceId: String = typeName + "-" + self.path.name
+  class QualifiedCounter(typeName: String, id: String) extends Counter(id) {
+    override def persistenceId: String = typeName + "-" + id
   }
 
-  class AnotherCounter extends QualifiedCounter("AnotherCounter")
+  def anotherCounterPropsFactory: String ⇒ Props =
+    id ⇒ Props(new AnotherCounter(id))
+
+  class AnotherCounter(id: String) extends QualifiedCounter("AnotherCounter", id)
 
   //#supervisor
   class CounterSupervisor extends Actor {
-    val counter = context.actorOf(Props[Counter], "theCounter")
+    val counter = context.actorOf(Props(new Counter("theCounter")), "theCounter")
```

Comment:
Props should be defined in companion object according to our recommendations, e.g. avoid mistakes of closing over non thread safe things such as `sender()`